eva_sub_cli/validators/validator.py

```diff
@@ -558,17 +558,38 @@ def _collect_file_info_to_metadata(self):
                                            file_name_2_md5.get(file_dict.get('fileName')) or ''
                         file_dict['fileSize'] = file_path_2_file_size.get(file_path) or \
                                                 file_name_2_file_size.get(file_dict.get('fileName')) or ''
+
+                        if not file_dict.get('fileSize'):
+                            error_txt = f"File size is not available for {file_dict.get('fileName')}"
+                            self.error(error_txt)
+                            errors.append({'property': '/files/fileSize', 'description': error_txt})
+                        if not file_dict.get('md5'):
+                            error_txt = f"md5 is not available for {file_dict.get('fileName')}"
+                            self.error(error_txt)
+                            errors.append({'property': '/files/md5', 'description': error_txt})
                         file_rows.append(file_dict)
                 else:
-                    self.error('No file found in metadata and multiple analysis alias exist: '
-                               'cannot infer the relationship between files and analysis alias')
+                    error_txt = ('No file section found in metadata and multiple analysis alias exist: '
+                                 'cannot infer the relationship between files and analysis alias')
+                    self.error(error_txt)
+                    errors.append({'property': '/files', 'description': error_txt})
                 metadata.set_files(file_rows)
             except Exception as e:
+                raise e
                 # Skip adding the md5
-                self.error('Error while loading or parsing metadata json: ' + str(e))
+                error_txt = 'Error while loading or parsing metadata json: ' + str(e)
+                self.error(error_txt)
+                errors.append({'property': '/', 'description': error_txt})
             metadata.write(self.metadata_json_post_validation)
         else:
-            self.error(f'Cannot locate the metadata in JSON format in {os.path.join(self.output_dir, "metadata.json")}')
+            error_txt = f'Cannot locate the metadata in JSON format in {os.path.join(self.output_dir, "metadata.json")}'
+            self.error(error_txt)
+            errors.append({'property': '/', 'description': error_txt})
+        if errors:
+            if 'json_errors' in self.results[METADATA_CHECK]:
+                self.results[METADATA_CHECK]['json_errors'].extend(errors)
+            else:
+                self.results[METADATA_CHECK]['json_errors'] = errors
```

Comment:
Won't it be confusing for users to see JSON errors in the report when they've only submitted a spreadsheet?

I can think of a couple places that seem more natural (the Project Details section, or even VCF checks for per-file reporting), but I guess those would be more complicated to add, and would basically require making file_info an independent validation check that can pass or fail, which seems like overkill for something that should almost always succeed... maybe I've convinced myself that reporting in the JSON errors is okay...